src/App.tsx

```diff
@@ -15,37 +15,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortOptions = {
+  field: 'alphabetical' | 'length' | null;
+  isReverse: boolean;
+};
+
+const prepareGoods = (
+  goods: string[],
+  { field, isReverse }: SortOptions,
+): string[] => {
+  const sortedGoods = [...goods];
+
+  if (field === 'alphabetical') {
+    sortedGoods.sort();
+  } else if (field === 'length') {
+    sortedGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortField, setSortField] = useState<'alphabetical' | 'length' | null>(
+    null,
+  );
+  const [originalGoods] = useState<string[]>(goodsFromServer);
+
+  const handleSortByName = (): void => {
+    setSortField('alphabetical');
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, {
+        field: 'alphabetical',
+        isReverse: isReversed,
+      }),
+    );
+  };
+
+  const handleSortByLength = (): void => {
+    setSortField('length');
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, { field: 'length', isReverse: isReversed }),
+    );
+  };
+
+  const handleReverse = (): void => {
+    setIsReversed(prev => !prev);
+    setVisibleGoods(
+      prepareGoods(goodsFromServer, {
+        field: sortField,
+        isReverse: !isReversed,
+      }),
+    );
+  };
+
+  const handleReset = (): void => {
+    setVisibleGoods(originalGoods);
```

Comment:
When resetting, you should consider setting `visibleGoods` to a sorted version of `originalGoods` based on the current sort settings, rather than directly setting it to `originalGoods`. This will ensure that the UI remains consistent with the current sort settings even after a reset.